gamification/src/main/java/juja/microservices/gamification/user/UserRepositoryCustom.java

```diff
@@ -29,10 +29,9 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
-package juja.microservices.gamification.model.repository;
+package juja.microservices.gamification.user;
```

Comment:
@vkuchyn such custom implementation and naming is using by spring-data. 
Firstly we create custom implementation (without word 'custom' there would be exceptions)
Secondly let standard repository interface extend the custom one. Doing so combines the CRUD and custom functionality and makes it available to clients. 

http://stackoverflow.com/questions/17008947/whats-the-difference-between-spring-datas-mongotemplate-and-mongorepository

btw. later i am planning to refactor to use another way of opertation with mongo. for example using  Morphia lib